Gillian-JS/lib/Compiler/JS2JSIL_Compiler.ml

```diff
@@ -2134,73 +2134,73 @@ let rec translate_expr tr_ctx e :
   | JS_Parser.Syntax.Call (e_f, xes)
     when e_f.JS_Parser.Syntax.exp_stx
          = JS_Parser.Syntax.Var js_symbolic_constructs.js_symb ->
-      let x =
-        match List.map (fun xe -> xe.JS_Parser.Syntax.exp_stx) xes with
-        | [ JS_Parser.Syntax.Var x ] -> "#" ^ x
+      let () =
+        match xes with
+        | [] -> ()
         | _ -> raise (Failure "Invalid symbolic")
       in
       let x_v = fresh_var () ^ "_v" in
-      let cmd1 = (metadata, None, LBasic (BCmd.Assignment (x_v, LVar x))) in
-      let cmd2 = (metadata, None, LLogic (LCmd.SpecVar [ x ])) in
+      let cmd1 = (metadata, None, LLogic (LCmd.FreshSVar x_v)) in
+      let x_v = PVar x_v in
```

Comment:
So we're definitely not checking if `x_v` is actually a `PVar` name, rather than, say `#imapvar`?

Reply:
So it can't be. In that case, that corresponds to a JS function call, so it has to be a JS program variable